npreadtext/_readers.py

```diff
@@ -110,8 +110,9 @@ def read(file, *, delimiter=',', comment='#', quote='"',
     # This will raise a LookupError if the encoding is unknown.
     codecs.lookup(encoding)
 
-    if dtype is not None and not isinstance(dtype, np.dtype):
-        dtype = np.dtype(dtype)
+    if dtype is None:
+        raise TypeError("a dtype must be provided.")
+    dtype = np.dtype(dtype)
```

Comment:
I went through the tests, a lot of unmodified tests should still be reasonable if `dtype=np.float64` implictly.